### PR TITLE
Pages custom Actions workflow

### PR DESCRIPTION
**Pull request by abernier**
READY TO MERGE

Provide a default `pages.yml` workflow that automatically deploys on GitHub Pages (if enabled in settings)

with the right `basePath` automatically

Demo: https://github.com/abernier/tailwind-nextjs-starter-blog/actions/runs/9870682135

- [x] replace testing-purpose values
- [x] Updated doc

### CODE REVIEW

File: .github/workflows/pages.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        type: string
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          repository: abernier/tailwind-nextjs-starter-blog
+          ref: pages
```
Comment (abernier) [inline .github/workflows/pages.yml:19]: I'm using my fork/branch here for testing purposes, but will obviously become:

```suggestion
          repository: timlrx/tailwind-nextjs-starter-blog
          ref: main
```

File: next.config.js [hunk 2]
```diff
@@ -54,12 +54,17 @@ const securityHeaders = [
   },
 ]
 
+const output = process.env.OUTPUT || undefined
+const basePath = process.env.BASE_PATH || undefined
```
Comment (abernier) [inline next.config.js:58]: defined at build time https://github.com/timlrx/tailwind-nextjs-starter-blog/pull/958/files#diff-267452bb53d388a7b14551d9fc53a797edc124dc9085ecc79f5339277a1e53b5R56-R57

File: next.config.js [hunk 3]
```diff
@@ -54,12 +54,18 @@ const securityHeaders = [
   },
 ]
 
+const output = process.env.EXPORT ? 'export' : undefined
+const basePath = process.env.BASE_PATH || undefined
+const unoptimized = process.env.UNOPTIMIZED ? true : undefined
```
Comment (abernier) [inline next.config.js:59]: defined at build time (by the workflow): `undefined` otherwise (inert)

File: .github/workflows/pages.yml [hunk 4]
```diff
@@ -0,0 +1,55 @@
+# Inspired from https://github.com/actions/starter-workflows/blob/main/pages/nextjs.yml
+name: GitHub Pages
+
+on:
+  push:
+    branches: pages
```
Comment (abernier) [inline .github/workflows/pages.yml:6]:
```suggestion
    branches: main
```

Comment (abernier) [reply .github/workflows/pages.yml:6]: i'm using `pages` branch here for testing purposes only

File: .github/workflows/pages.yml [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+          path: .next/cache
+          key: ${{ runner.os }}-nextjs-${{ hashFiles('**/yarn.lock') }}-${{ hashFiles('**.[jt]s', '**.[jt]sx') }}
+          restore-keys: |
+            ${{ runner.os }}-nextjs-${{ hashFiles('**/yarn.lock') }}-
+      - run: yarn
+      - run: yarn build
+        env:
+          EXPORT: 1
+          UNOPTIMIZED: 1
+          BASE_PATH: ${{ steps.configurepages.outputs.base_path }}
```
Comment (abernier) [inline .github/workflows/pages.yml:41]: `basePath` from `actions/configure-pages@v5` 🥳

**Comment by timlrx** [comment]
Thanks! This greatly simplifies deploying to Github pages out of the box and also thanks for updating the readme. Let's give it a try!

**Comment by abernier** [comment]
do not hesitate to enable it in project's settings ;)

nb: could be great adding a link to timlrx.github.io/tailwind-nextjs-starter-blog in the readme ;)

**Comment by timlrx** [comment]
I have merged it in and almost everything works except for the cmd-k search. I believe it is probably a base path issue. Do you mind taking a look and once you are done, you can also include the demo link in the doc: https://timlrx.github.io/tailwind-nextjs-starter-blog/

**Comment by abernier** [comment]
> except for the cmd-k search

https://github.com/timlrx/tailwind-nextjs-starter-blog/pull/962 fixes this

**Pull request merged**